File: src/main/java/com/bot/db/mappers/RssSubscriptionMapper.java
package com.bot.db.mappers;

import com.bot.models.RssProvider;
import com.bot.models.RssSubscription;

import java.sql.ResultSet;
import java.sql.SQLException;

public class RssSubscriptionMapper {
    public static RssSubscription mapSetToRssSubscription(ResultSet set) throws SQLException {
        return new RssSubscription(
                set.getInt("id"),
                set.getString("subject"),
                RssProvider.getProvider(set.getInt("provider")),
                set.getBoolean("nsfw")
        );
    }
}

File: src/main/java/com/bot/db/mappers/RssSubscriptionMapper.kt
package com.bot.db.mappers

import com.bot.models.RssProvider.Companion.getProvider
import com.bot.models.RssSubscription
import java.sql.ResultSet
import java.sql.SQLException

object RssSubscriptionMapper {
    @JvmStatic
    @Throws(SQLException::class)
    fun mapSetToRssSubscription(set: ResultSet): RssSubscription {
        return RssSubscription(
                set.getInt("id"),
                set.getString("subject"),
                getProvider(set.getInt("provider")),
                set.getBoolean("nsfw")
        )
    }
}